dpnp/tests/test_nanfunctions.py

```diff
@@ -505,86 +501,78 @@ class TestNanProd:
     @pytest.mark.parametrize("axis", [None, 0, 1, -1, 2, -2, (1, 2), (0, -2)])
     @pytest.mark.parametrize("keepdims", [False, True])
     @pytest.mark.parametrize("dtype", get_float_complex_dtypes())
-    def test_nanprod(self, axis, keepdims, dtype):
-        a = numpy.arange(1, 13, dtype=dtype).reshape((2, 2, 3))
+    def test_basic(self, axis, keepdims, dtype):
+        a = generate_random_numpy_array((2, 3, 4), dtype=dtype)
         a[:, :, 2] = numpy.nan
         ia = dpnp.array(a)
 
-        np_res = numpy.nanprod(a, axis=axis, keepdims=keepdims)
-        dpnp_res = dpnp.nanprod(ia, axis=axis, keepdims=keepdims)
+        expected = numpy.nanprod(a, axis=axis, keepdims=keepdims)
+        result = dpnp.nanprod(ia, axis=axis, keepdims=keepdims)
 
-        assert dpnp_res.shape == np_res.shape
-        assert_allclose(dpnp_res, np_res)
+        assert result.shape == expected.shape
+        assert_allclose(result, expected, rtol=1e-6)
 
     @pytest.mark.usefixtures("suppress_complex_warning")
     @pytest.mark.usefixtures("suppress_invalid_numpy_warnings")
-    @pytest.mark.parametrize("in_dtype", get_float_complex_dtypes())
-    @pytest.mark.parametrize(
-        "out_dtype", get_all_dtypes(no_bool=True, no_none=True)
-    )
-    def test_nanprod_dtype(self, in_dtype, out_dtype):
-        a = numpy.arange(1, 13, dtype=in_dtype).reshape((2, 2, 3))
+    @pytest.mark.parametrize("in_dt", get_float_complex_dtypes())
+    @pytest.mark.parametrize("dt", get_all_dtypes(no_bool=True, no_none=True))
```

Comment:
Should we allow `bool` and `None` here, because the test is intended for testing different dtype values passed to nanprod?
```suggestion
    @pytest.mark.parametrize("dt", get_all_dtypes()
```